platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/components/FutureEventBufferWiring.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.wiring.components;
+
+import com.swirlds.common.wiring.schedulers.TaskScheduler;
+import com.swirlds.common.wiring.wires.input.BindableInputWire;
+import com.swirlds.common.wiring.wires.input.InputWire;
+import com.swirlds.common.wiring.wires.output.OutputWire;
+import com.swirlds.platform.consensus.NonAncientEventWindow;
+import com.swirlds.platform.event.FutureEventBuffer;
+import com.swirlds.platform.event.GossipEvent;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.List;
+
+/**
+ * The wiring for the {@link com.swirlds.platform.event.FutureEventBuffer}.
+ *
+ * @param eventInput       an input wire with events in topological order, possibly containing time travelers
+ * @param eventWindowInput an output wire with events in topological order, guaranteed no time travelers
+ * @param flushRunnable    a runnable that will flush the future event buffer
+ */
+public record FutureEventBufferWiring(
+        @NonNull InputWire<GossipEvent> eventInput,
+        @NonNull InputWire<NonAncientEventWindow> eventWindowInput,
+        @NonNull OutputWire<GossipEvent> eventOutput,
+        @NonNull Runnable flushRunnable) {
+
+    /**
+     * Create a new instance of the FutureEventBufferWiring.
+     *
+     * @return a new instance of the FutureEventBufferWiring
+     */
+    @NonNull
+    public static FutureEventBufferWiring create(@NonNull final TaskScheduler<List<GossipEvent>> scheduler) {
+        final InputWire<GossipEvent> eventInput = scheduler.buildInputWire("preconsensus events");
+        final InputWire<NonAncientEventWindow> eventWindowInput = scheduler.buildInputWire("non-ancient event window");
+
+        final OutputWire<List<GossipEvent>> eventListOutputWIre = scheduler.getOutputWire();
+        final OutputWire<GossipEvent> eventOutputWire =
+                eventListOutputWIre.buildSplitter("futureEventBufferSplitter", "possible parent lists");
+
+        final Runnable flushRunnable = scheduler::flush;
+
+        return new FutureEventBufferWiring(eventInput, eventWindowInput, eventOutputWire, flushRunnable);
+    }
+
+    /**
+     * Bind the input and output wires to the future event buffer.
```

Comment:
output wires aren't being bound

Reply:
only input wires need to be bound, you never bind the output wires

Reply:
Right, but the method description says otherwise

"Bind the input and output wires"